alacritty_terminal/src/locale.rs

```diff
@@ -59,18 +58,13 @@ pub fn set_locale_environment() {
     let locale_c_str = CString::new(locale_id.to_owned()).unwrap();
     let locale_ptr = locale_c_str.as_ptr();
     let locale_id = unsafe {
-        // save a copy of original setting
-        let original = setlocale(LC_CTYPE, null());
-        let saved_original = if original.is_null() {
-            CString::new("").unwrap()
-        } else {
-            CStr::from_ptr(original).to_owned()
-        };
         // try setting `locale_id`
         let modified = setlocale(LC_CTYPE, locale_ptr);
         let result = if modified.is_null() { String::new() } else { locale_id };
-        // restore original setting
-        setlocale(LC_CTYPE, saved_original.as_ptr());
+        // restore LC_CTYPE to UTF-8
+        let utf8 = CString::new("UTF-8").unwrap();
```

Comment:
If we decide to fallback to some arbitrary value, could we use `C.UTF-8` instead of `UTF-8`, because the later one isn't a valid locale if you ssh into some linux(glibc at least) box?

Reply:
We could. My only reason to choose `UTF-8` was because that is how Terminal.app and iTerm does it.